### ISSUE TITLE
docker deployment microphone and webcam not working on rocky linux/ rhel 9 VM

### ISSUE THREAD

**Issue by leonidas-o** [bigbluebutton/docker#246]
Running on 
Rocky Linux VERSION="9.1 (Blue Onyx)"

Installed docker and docker compose like described in: https://docs.docker.com/engine/install/centos/
> added repo and `dnf install docker-ce docker-ce-cli containerd.io docker-compose-plugin`
> Docker Compose version v2.14.1

So I'm not using `docker-compose` but `docker compose`.
For the rest, I've done the steps 2 - 6: https://github.com/bigbluebutton/docker

Tried with the main and the develop branch, both behave the same, both are not working for me.
After `docker compose up -d` the containers are up and healthy.

```
CONTAINER ID   IMAGE                                            COMMAND                  CREATED         STATUS                   PORTS                   NAMES
b958abb671cc   alangecker/bbb-docker-nginx:1.21-v4.0.0-v2.5.5   "/docker-entrypoint.…"   8 minutes ago   Up 8 minutes                                     bbb-docker-nginx-1
d4e519d95a13   alangecker/bbb-docker-recordings:v2.5.7          "/bin/sh -c /entrypo…"   8 minutes ago   Up 8 minutes                                     bbb-docker-recordings-1
20846d6ea027   alangecker/bbb-docker-html5:v2.5.7               "/entrypoint.sh"         8 minutes ago   Up 8 minutes                                     bbb-docker-html5-backend-1-1
2a72fb47a275   alangecker/bbb-docker-web:v2.5.8                 "/entrypoint.sh"         8 minutes ago   Up 8 minutes (healthy)                           bbb-docker-bbb-web-1
a1edd2dfa5ee   alangecker/bbb-docker-html5:v2.5.7               "/entrypoint.sh"         8 minutes ago   Up 8 minutes                                     bbb-docker-html5-backend-2-1
0ea7ad298163   alangecker/bbb-docker-html5:v2.5.7               "/entrypoint.sh"         8 minutes ago   Up 8 minutes                                     bbb-docker-html5-frontend-1-1
30f4a2b25ec0   alangecker/bbb-docker-html5:v2.5.7               "/entrypoint.sh"         8 minutes ago   Up 8 minutes                                     bbb-docker-html5-frontend-2-1
b3a0ccb88aa8   alangecker/bbb-docker-pads:v1.2.2                "/bin/sh -c /entrypo…"   8 minutes ago   Up 8 minutes                                     bbb-docker-bbb-pads-1
7255292adafd   alangecker/bbb-docker-etherpad:1.8.18-3          "/entrypoint.sh"         8 minutes ago   Up 8 minutes (healthy)   9001/tcp                bbb-docker-etherpad-1
89f8fd0c1ff7   alangecker/bbb-docker-apps-akka:v2.5.7           "/bin/sh -c /entrypo…"   8 minutes ago   Up 8 minutes                                     bbb-docker-apps-akka-1
09b6f104b26a   alangecker/bbb-docker-webrtc-sfu:v2.8.6          "docker-entrypoint.s…"   8 minutes ago   Up 8 minutes                                     bbb-docker-webrtc-sfu-1
f18c6246a866   alangecker/bbb-docker-fsesl-akka:v2.5.7          "/bin/sh -c /entrypo…"   8 minutes ago   Up 8 minutes                                     bbb-docker-fsesl-akka-1
1123e87f4b15   alangecker/bbb-docker-periodic:v2.5.0-rc.1       "/entrypoint.sh"         8 minutes ago   Up 8 minutes                                     bbb-docker-periodic-1
c484ed9d22fb   bigbluebutton/greenlight:v2                      "bin/start"              8 minutes ago   Up 8 minutes             10.7.7.1:5000->80/tcp   bbb-docker-greenlight-1
3cd327227d98   alangecker/bbb-docker-jodconverter:latest        "/docker-entrypoint.…"   8 minutes ago   Up 8 minutes                                     bbb-docker-jodconverter-1
45a589cad789   kurento/kurento-media-server:6.18                "/entrypoint.sh"         8 minutes ago   Up 8 minutes (healthy)                           bbb-docker-kurento-1
c70508fc4da0   postgres:12-alpine                               "docker-entrypoint.s…"   8 minutes ago   Up 8 minutes (healthy)   5432/tcp                bbb-docker-postgres-1
8f879c5bd653   alangecker/bbb-docker-freeswitch:v2.5.7          "/bin/sh -c /entrypo…"   8 minutes ago   Up 8 minutes                                     bbb-freeswitch
2fde6e5ce493   mongo:4.4                                        "docker-entrypoint.s…"   8 minutes ago   Up 8 minutes (healthy)   27017/tcp               bbb-mongodb
cd54aa15f657   redis:7.0-alpine                                 "docker-entrypoint.s…"   8 minutes ago   Up 8 minutes (healthy)   6379/tcp                bbb-docker-redis-1
```


`ss -tulpn` shows the following:
```
Netid                           State                             Recv-Q                            Send-Q                                                       Local Address:Port                                                        Peer Address:Port                           Process                                                                                                                                                                                                                                                                                
udp                             UNCONN                            0                                 0                                                                 10.7.7.1:5060                                                             0.0.0.0:*                               users:(("freeswitch",pid=859484,fd=39))                                                                                                                                                                                                                                               
udp                             UNCONN                            0                                 0                                                                 10.7.7.1:5090                                                             0.0.0.0:*                               users:(("freeswitch",pid=859484,fd=48))                                                                                                                                                                                                                                               
udp                             UNCONN                            0                                 0                                                                127.0.0.1:323                                                              0.0.0.0:*                               users:(("chronyd",pid=812,fd=5))                                                                                                                                                                                                                                                      
udp                             UNCONN                            0                                 0                                                                    [::1]:5060                                                                [::]:*                               users:(("freeswitch",pid=859484,fd=28))                                                                                                                                                                                                                                               
udp                             UNCONN                            0                                 0                                                                    [::1]:5090                                                                [::]:*                               users:(("freeswitch",pid=859484,fd=34))                                                                                                                                                                                                                                               
udp                             UNCONN                            0                                 0                                                                    [::1]:323                                                                 [::]:*                               users:(("chronyd",pid=812,fd=6))                                                                                                                                                                                                                                                      
tcp                             LISTEN                            0                                 511                                                               10.7.7.1:3008                                                             0.0.0.0:*                               users:(("node",pid=860299,fd=22))                                                                                                                                                                                                                                                     
tcp                             LISTEN                            0                                 511                                                              127.0.0.1:3010                                                             0.0.0.0:*                               users:(("node",pid=860516,fd=60))                                                                                                                                                                                                                                                     
tcp                             LISTEN                            0                                 64                                                                10.7.7.1:5090                                                             0.0.0.0:*                               users:(("freeswitch",pid=859484,fd=50))                                                                                                                                                                                                                                               
tcp                             LISTEN                            0                                 64                                                                10.7.7.1:5060                                                             0.0.0.0:*                               users:(("freeswitch",pid=859484,fd=40))                                                                                                                                                                                                                                               
tcp                             LISTEN                            0                                 511                                                              127.0.0.1:3014                                                             0.0.0.0:*                               users:(("node",pid=860516,fd=64))                                                                                                                                                                                                                                                     
tcp                             LISTEN                            0                                 511                                                              127.0.0.1:3016                                                             0.0.0.0:*                               users:(("node",pid=860299,fd=24))                                                                                                                                                                                                                                                     
tcp                             LISTEN                            0                                 4096                                                              10.7.7.1:5000                                                             0.0.0.0:*                               users:(("docker-proxy",pid=859434,fd=4))                                                                                                                                                                                                                                              
tcp                             LISTEN                            0                                 64                                                               127.0.0.1:5066                                                             0.0.0.0:*                               users:(("freeswitch",pid=859484,fd=45))                                                                                                                                                                                                                                               
tcp                             LISTEN                            0                                 64                                                                10.7.7.1:5066                                                             0.0.0.0:*                               users:(("freeswitch",pid=859484,fd=44))                                                                                                                                                                                                                                               
tcp                             LISTEN                            0                                 64                                                              172.27.0.1:5066                                                             0.0.0.0:*                               users:(("freeswitch",pid=859484,fd=43))                                                                                                                                                                                                                                               
tcp                             LISTEN                            0                                 64                                                              172.17.0.1:5066                                                             0.0.0.0:*                               users:(("freeswitch",pid=859484,fd=42))                                                                                                                                                                                                                                               
tcp                             LISTEN                            0                                 64                                                               10.1.1.57:5066                                                             0.0.0.0:*                               users:(("freeswitch",pid=859484,fd=38))                                                                                                                                                                                                                                               
tcp                             LISTEN                            0                                 64                                                                10.7.7.1:7443                                                             0.0.0.0:*                               users:(("freeswitch",pid=859484,fd=49))                                                                                                                                                                                                                                               
tcp                             LISTEN                            0                                 5                                                                 10.7.7.1:8021                                                             0.0.0.0:*                               users:(("freeswitch",pid=859484,fd=56))                                                                                                                                                                                                                                               
tcp                             LISTEN                            0                                 128                                                                0.0.0.0:22                                                               0.0.0.0:*                               users:(("sshd",pid=954,fd=3))                                                                                                                                                                                                                                                         
tcp                             LISTEN                            0                                 511                                                                0.0.0.0:48087                                                            0.0.0.0:*                               users:(("nginx",pid=861592,fd=8),("nginx",pid=861591,fd=8),("nginx",pid=861590,fd=8),("nginx",pid=861583,fd=8),("nginx",pid=861582,fd=8),("nginx",pid=861581,fd=8),("nginx",pid=861579,fd=8),("nginx",pid=861578,fd=8),("nginx",pid=861488,fd=8))                                     
tcp                             LISTEN                            0                                 64                                                                   [::1]:5090                                                                [::]:*                               users:(("freeswitch",pid=859484,fd=35))                                                                                                                                                                                                                                               
tcp                             LISTEN                            0                                 64                                                                   [::1]:5060                                                                [::]:*                               users:(("freeswitch",pid=859484,fd=29))                                                                                                                                                                                                                                               
tcp                             LISTEN                            0                                 64                                                                   [::1]:5066                                                                [::]:*                               users:(("freeswitch",pid=859484,fd=30))                                                                                                                                                                                                                                               
tcp                             LISTEN                            0                                 64                                                                   [::1]:7443                                                                [::]:*                               users:(("freeswitch",pid=859484,fd=31))                                                                                                                                                                                                                                               
tcp                             LISTEN                            0                                 128                                                                   [::]:22                                                                  [::]:*                               users:(("sshd",pid=954,fd=4))                                                                                                                                                                                                                                                         
tcp                             LISTEN                            0                                 511                                                                   [::]:48087                                                               [::]:*                               users:(("nginx",pid=861592,fd=9),("nginx",pid=861591,fd=9),("nginx",pid=861590,fd=9),("nginx",pid=861583,fd=9),("nginx",pid=861582,fd=9),("nginx",pid=861581,fd=9),("nginx",pid=861579,fd=9),("nginx",pid=861578,fd=9),("nginx",pid=861488,fd=9))                                     
tcp                             LISTEN                            0                                 128                                                                      *:8888                                                                   *:*                               users:(("kurento-media-s",pid=859413,fd=17))                                                                                                                                                                                                                                          
```


Now when trying to access the webpage, it tries to load, gets stuck and after some time it shows: "Safari can't open the page ... because the server where this page is located isn't responding."
Nothing is listening on port 80, when looking at the `ss` command output.
Also tried with the external public IP when running `./scripts/setup` as well as the VM's IP. Anything special that has to be done when deploying to a VM?

**Comment by leonidas-o** [comment]
okay, found this doc: https://github.com/bigbluebutton/docker/blob/c6b654863e885237e0cfcc718bf3d6b9c483b0a3/docs/existing-web-server.md

and saw that it is using the port `48087`. Now I could make it to serve the web UI at least.

Nevertheless there are still issues, the microphone and webcam sharing is not working. I've executed the check, which looks good except the warning.

`docker exec -it bbb-docker-greenlight-1 bundle exec rake conf:check`
```bash
warning: parser/current is loading parser/ruby27, which recognizes2.7.6-compliant syntax, but you are running 2.7.5.
Please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.

Checking environment: Passed
Checking Connection: Passed
Checking Secret: Passed
Checking SMTP connection: Passed
```

When I click on `Microphone` ("How would you like to join the audio"), I get the following: `Failure on call (reason=ICE error) (error 1004)`

When I click on share webcam, it shows my webcam, camera and quality settings, I can see myself, but then on `start sharing`, I get: `Media server failed to process request (error 2200)`

I also tried Safari and Firefox.

**Comment by leonidas-o** [comment]
Couldn't make BigBlueButton work behind a NAT, with opnsense/pfsense 1:1 NAT + Reflection etc., so I gave up on that approach. I still found a solution assigning the second public IP to the BBB VM, which I documented here: https://serverfault.com/questions/1121061/assigned-second-public-ip-to-vm-from-outside-not-reachable/1121266#1121266

**Comment by killmasta93** [comment]
hi @leonidas-o currently having the same issue, but i currenly have hosted proxmox onpremise and not on hetzner? what would be the correct way?

**Comment by leonidas-o** [comment]
@killmasta93 what's the difference? I'm not using hetzner cloud but a hetzner dedicated server with proxmox on it. So the description in the link I posted above should work for you, too. The hetzner docs I linked there can also be used for other providers, more or less they all have basically the same features.
(But if your environment is a home lab without a static public IP, I'm not sure if that will work.)
